src/jimple-frontend/AST/jimple_class_method.cpp

```diff
@@ -0,0 +1,136 @@
+#include <jimple-frontend/AST/jimple_class_member.h>
+#include <util/std_code.h>
+#include <util/expr_util.h>
+#include <util/message/format.h>
+
+exprt jimple_class_method::to_exprt(
+  contextt &ctx,
+  const std::string &class_name,
+  const std::string &) const
+{
+  // Dummy will be the base to be returned
+  exprt dummy;
+  code_typet method_type;
+  typet inner_type;
+  inner_type = t.to_typet();
+  method_type.return_type() = inner_type;
+
+  auto id = get_method_name(class_name, name);
+  auto symbol = create_jimple_symbolt(method_type, class_name, name, id);
+
+  std::string symbol_name = symbol.id.as_string();
+
+  symbol.lvalue = true;
+  symbol.is_extern = false;
+  symbol.file_local = true;
+
+  ctx.move_symbol_to_context(symbol);
+  symbolt &added_symbol = *ctx.find_symbol(symbol_name);
+
+  // In jimple, every non-static method has access to @this
+  // In future, I will add this as a paremeter to the function call
+  if(!m.is_static())
+  {
+    auto this_type = int_type(); // TODO: support the struct type
+    std::string param_id, param_name;
+
+    std::ostringstream oss;
+    oss << "@this";
+    auto temp = get_symbol_name(class_name, name, oss.str());
+    param_id = temp;
+    param_name = oss.str();
+
+    auto param_symbol =
+      create_jimple_symbolt(this_type, class_name, param_name, param_id, id);
+    param_symbol.lvalue = true;
+    param_symbol.is_parameter = true;
+    param_symbol.file_local = true;
+
+    ctx.move_symbol_to_context(param_symbol);
+  }
+
+  // In Jimple, every parameter is named as @parameter#
+  for(long unsigned int i = 0; i < parameters.size(); i++)
+  {
+    auto param_type = parameters[i]->to_typet();
+
+    code_typet::argumentt param;
+
+    std::string param_id, param_name;
+
+    std::ostringstream oss;
+    oss << "@parameter" << i;
+    auto temp = get_symbol_name(class_name, name, oss.str());
+    param_id = temp;
+    param_name = oss.str();
+    param = code_typet::argumentt();
+    param.type() = param_type;
+    param.cmt_base_name(param_name);
+    param.cmt_identifier(param_id);
+
+    auto param_symbol =
+      create_jimple_symbolt(param_type, class_name, param_name, param_id, id);
+    param_symbol.lvalue = true;
+    param_symbol.is_parameter = true;
+    param_symbol.file_local = true;
+
+    ctx.move_symbol_to_context(param_symbol);
+    method_type.arguments().push_back(param);
+  }
+
+  // Apparently, if the type has no arguments, we assume ellipsis
+  if(!method_type.arguments().size())
+    method_type.make_ellipsis();
```

Comment:
Do we really support it? See https://github.com/esbmc/esbmc/issues/584.

Reply:
From https://github.com/esbmc/esbmc/pull/582#discussion_r781453611:
> We do support variadic functions, but only on 32 bits.

Reply:
I am not sure how to handle though. Should we ignore this then?